File: typescript/src/user_agent_validation_test.ts
import OnboardingBuddyClient from '@onboardingbuddy/onboardingbuddy-client'
import * as dotenv from 'dotenv'
dotenv.config()

const obAppKey = process.env.OB_APP_KEY || '';
const obApiKey = process.env.OB_API_KEY || '';
const obApiSecret = process.env.OB_API_SECRET || '';

const client = new OnboardingBuddyClient(obAppKey, obApiKey, obApiSecret);

const browserValidation = client.validateBrowser({
  userAgent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36',
});

browserValidation
    .then(response => 
        { 
            console.log(`User Agent: ${response.data.userAgent}`)
            console.log(`Simple Software: ${response.data.simpleSoftware}`)
            console.log(`Software: ${response.data.software}`)
            console.log(`Software Name: ${response.data.softwareName}`)
            console.log(`Operating System: ${response.data.operatingSystem}`)
            console.log(`Operating System Flavour: ${response.data.operatingSystemFlavour}`)
            console.log(`Operating System Version: ${response.data.operatingSystemVersion}`)
            console.log(`Is Abusive: ${response.data.isAbusive}`)
            console.log(`Check Status: ${response.data.checkStatus}`)
            console.log(`Message Id: ${response.data.messageId}`)
        })
    .catch(error=> console.log(error.response.data));

File: python/user_agent_validation_test.py
from onboarding_buddy_client.api.validation_api import ValidationApi
from onboarding_buddy_client.models.browser_request_m import BrowserRequestM
from onboarding_buddy_client import Configuration, ApiClient
from dotenv import load_dotenv
import os

load_dotenv()

# Configure the client with authentication headers
config = Configuration(
    api_key={
        "AppKey": os.getenv("OB_APP_KEY"),
        "ApiKey": os.getenv("OB_API_KEY"),
        "ApiSecret": os.getenv("OB_API_SECRET")
    }
)

# Initialize the API client
api_client = ApiClient(configuration=config)
validation_api = ValidationApi(api_client)

try:
    request = BrowserRequestM(
        user_agent="Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
    )

    response = validation_api.browser(browser_request_m=request)
    
    print(f"User Agent: {response.user_agent}")
    print(f"Simple Software: {response.simple_software}")
    print(f"Software: {response.software}")
    print(f"Software Name: {response.software_name}")
    print(f"Operating System: {response.operating_system}")
    print(f"Operating System Flavour: {response.operating_system_flavour}")
    print(f"Operating System Version: {response.operating_system_version}")
    print(f"Is Abusive: {response.is_abusive}")
    print(f"Check Status: {response.check_status}")
    print(f"Message Id: {response.message_id}")

except Exception as e:
    print(f"Error: {e}")